pkgs/esp-idf/default.nix

```diff
@@ -93,7 +93,7 @@ stdenv.mkDerivation rec {
 
   # Do not preserve file modes when copying templates as the nix store is read-only.
   # create-project will fail without this.
-  patches = [ ./template-modes.patch ];
+  patches = [ ];
```

Comment:
Since you are removing the patch, you may also want to remove the comment above.